ui/components/app/connected-sites-list/connected-snaps.js

```diff
@@ -0,0 +1,132 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { useHistory } from 'react-router-dom';
+import { WALLET_SNAP_PERMISSION_KEY } from '@metamask/snaps-utils';
+import { useDispatch, useSelector } from 'react-redux';
+import { SnapCaveatType } from '@metamask/rpc-methods';
+import { Box, IconName, IconSize, Text } from '../../component-library';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import { MenuItem } from '../../ui/menu';
+import { SNAPS_VIEW_ROUTE } from '../../../helpers/constants/routes';
+import SnapAvatar from '../snaps/snap-avatar';
+import {
+  AlignItems,
+  BlockSize,
+  Display,
+  FlexDirection,
+  JustifyContent,
+  TextVariant,
+} from '../../../helpers/constants/design-system';
+import ConnectedAccountsListOptions from '../connected-accounts-list/connected-accounts-list-options';
+import {
+  getOriginOfCurrentTab,
+  getPermissionSubjects,
+} from '../../../selectors';
+import { removePermissionsFor, updateCaveat } from '../../../store/actions';
+
+export default function ConnectedSnaps({ connectedSubjects }) {
+  const [showOptions, setShowOptions] = useState();
+  const t = useI18nContext();
+  const history = useHistory();
+  const dispatch = useDispatch();
+  const subjects = useSelector((state) => getPermissionSubjects(state));
+  const connectedOrigin = useSelector(getOriginOfCurrentTab);
+
+  const onDisconnect = (snapId) => {
+    const caveatValue =
+      subjects[connectedOrigin].permissions[WALLET_SNAP_PERMISSION_KEY]
+        .caveats[0].value;
+    const newCaveatValue = { ...caveatValue };
+    delete newCaveatValue[snapId];
+    if (Object.keys(newCaveatValue).length > 0) {
+      dispatch(
+        updateCaveat(
+          connectedOrigin,
+          WALLET_SNAP_PERMISSION_KEY,
+          SnapCaveatType.SnapIds,
+          newCaveatValue,
+        ),
+      );
+    } else {
+      dispatch(
+        removePermissionsFor({
+          [connectedOrigin]: [WALLET_SNAP_PERMISSION_KEY],
+        }),
+      );
+    }
+  };
+
+  const settingsClick = (id) => {
+    history.push(`${SNAPS_VIEW_ROUTE}/${encodeURIComponent(id)}`);
+  };
+  const renderListItemOptions = (snapId) => {
```

Comment:
Any reason why this is a function and not just combined with the rest of the markup?

Reply:
Nope! This can be moved inside the Item. Updated here https://github.com/MetaMask/metamask-extension/pull/20811/commits/d852bc33d66b72d794e1f5a236ecc06b4f5d8854

Reply:
Oh, I actually meant `renderListItemOptions` 😅

Reply:
Aah! My bad. So, renderListItemOptions is using `ConnectedAccountsListOptions` which is the same component used to render options in connected-accounts. Hence I kept it in a separate function. It's the same way how we handled in `connected-accounts-list.component.js`